tools/src/main/scala/scala/scalanative/build/NativeConfig.scala

```diff
@@ -71,13 +71,43 @@ sealed trait NativeConfig {
   /** Shall we use the incremental compilation? */
   def useIncrementalCompilation: Boolean
 
-  /** Shall be compiled with multithreading support. If equal to `None` the
-   *  toolchain would detect if program uses system threads - when not thrads
-   *  are not used, the program would be linked without multihreading support.
+   // format: off
+  /** Shall be compiled with multithreading support.
+   *
+   *  'show ThisBuild/nativeConfig' will display one of three values.
+   *  The default 'detect' setting is appropriate for almost all cases.
+   *
+   *  * 'detect' - The toolchain will start linking using 'true'.
+   *               If by the end no use of system threads is found, the
+   *               toolchain will re-link using 'false' to reduce
+   *               synchronization overhead.
+   *
+   *               `nativeConfig` is initialized to this value and
+   *               there is no easy way to set it once changed.
+   *
+   *  * 'false'  - Never link with multithreading enabled.
+   *
+   *              Note Well:
+   *
+   *                + This setting is not tested in Scala Native Continuous
+   *                  integration.
+   *
+   *                + Some Scala Native classes, such as `java.lang.Process*`
+   *                  require multithreading and may fail to link with this
+   *                  setting.
+   *
+   *                + Code using `Future`s may link but encounter runtime
+   *                  problems.
+   *
+   *  * 'true'   - Always link with multithreading enabled.
+   * 
    */
+  // format: on
```

Comment:
I think in general there are ways to make this look nice without turning formatting off. In general that is not a great practice as it creates a burden on future maintainers. https://docs.scala-lang.org/overviews/scaladoc/for-library-authors.html

Reply:
Eric,

Thank you for looking at this.

I am off-line for a few days. Link you kindly provided gave me an idea of something to try next session.